lib/src/rules/prefer_is_empty.dart

```diff
@@ -111,13 +112,8 @@ class _Visitor extends SimpleAstVisitor<void> {
     }
 
     // Should be subtype of Iterable, Map or String.
-    TypeProvider typeProvider = context.typeProvider;
-    TypeSystem typeSystem = context.typeSystem;
-
-    if (typeSystem.mostSpecificTypeArgument(type, typeProvider.iterableType) ==
-            null &&
-        typeSystem.mostSpecificTypeArgument(type, typeProvider.mapType) ==
-            null &&
+    if (!DartTypeUtilities.implementsInterface(type, 'Iterable', 'dart.core') &&
+        !DartTypeUtilities.implementsInterface(type, 'Map', 'dart.core') &&
```

Comment:
I thought using `typeSystem` was the preferred way?

Reply:
In general - yes, this specific API - no.

It should be something like (currently non-API `InterfaceTypeImpl`) `InterfaceType asInstanceOf(ClassElement targetElement)`. Passing raw `Map` type is weird, we should use `ClassElement`.

Reply:
Thanks!